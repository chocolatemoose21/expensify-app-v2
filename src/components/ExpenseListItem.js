import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = ({ id, description, amount, createdAt}) => (
  <div>
    <Link className= "list-item" to={`/edit/${id}`}>
      <div>
        <p>{description}</p>
      </div>

    </Link>
    <h3>{amount}</h3>
    <p>{createdAt}</p>

  </div>
);

export default ExpenseListItem;
