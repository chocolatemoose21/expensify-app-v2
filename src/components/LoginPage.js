import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

//The functionality I want is when you click a button it goes there.

export const LoginPage = ({ startLogin }) => (

  <div className="box-layout">
    <div className = "box-layout__box">
      <h1 clasName = "box-layout__title ">Expensify</h1>
        <p> It's time to get your expenses under control</p>
        <button onClick={startLogin} className = "button"> Login With Google </button>
    </div>
  </div>
);

//This is the function to generate the action (i.e. dispatch the taxi)
const mapDispatchToProps = (dispatch) => ({
  startLogin: ()=> dispatch(startLogin())
});

//This is connecting the component to the dispact action.
export default connect(undefined, mapDispatchToProps)(LoginPage);
