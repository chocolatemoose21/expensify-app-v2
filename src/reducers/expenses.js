
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [
        ...state,
        action.expense
      ]
      //How do you remove an expense by id from state from an array? You use filter!
      //State.filter gives you an new array.
    case 'REMOVE_EXPENSE':
      return state.filter(({ id } ) => id !== action.id);
    case 'EDIT_EXPENSE':
      return state.map((expense)=> {
        if (expense.id === action.id){
          return{
            ...expense,
            ...action.updates
          }
        } else {
          return expense;
        }
      });
      default:
        return state;
  }
};

export default expensesReducer;
