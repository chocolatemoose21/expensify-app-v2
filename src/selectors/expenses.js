import moment from 'moment';

//Get Visible EXPENSES
export default (expenses, { text, sortBy, startDate, endDate, }) => {
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.CreatedAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day'): true;
    const endDateMatch = endDate ? endDate.isSameOrBefore(createdAtMoment,'day'): true;
    const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a,b)=>{
    if (sortBy === 'date'){
      return a.createdDate < b.createdAt ? 1 : -1;
    } else if (sortBy === 'amount'){
      return a.amount < b.amount ? 1:-1;
    }
  });
};
