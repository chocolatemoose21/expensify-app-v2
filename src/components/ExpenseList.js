import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    <div className= "content-container">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
    <h1> Expense List</h1>
    {props.expenses.map((expense)=>{
      return <ExpenseListItem key={expense.id} {...expense} />;
    })}
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
