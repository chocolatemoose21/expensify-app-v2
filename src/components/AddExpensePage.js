import React from 'react';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const AddExpensePage = (props) => (
  <div>
    <p> This is my add expense component </p>
    <ExpenseForm
      onSubmit = {(expense)=>{
        props.dispatch(startAddExpense(expense))
        props.history.push('/');
      }}
      />
  </div>
);

export default connect()(AddExpensePage);
