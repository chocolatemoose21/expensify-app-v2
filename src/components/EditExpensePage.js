import React from 'react';
import { connect } from 'react-redux';
import  ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
  return(
  <div>
    <div className = "content-container">
      <div>
        <h1> Edit Expense</h1>
      </div>

    <ExpenseForm
      expense = {props.expense}
      onSubmit={(expense) => {
        props.dispatch(editExpense(props.expense.id,expense));
        props.history.push('/')
      }}
      />
      <button
        className = "button--secondary"
        onClick = {(e) => {
          props.dispatch(removeExpense({id:props.expense.id}));
          props.history.push('/')
          }}>Remove</button>
      </div>
  </div>
)
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense)=>expense.id === props.match.params.id)
  };
};

export default connect(mapStateToProps)(EditExpensePage);
