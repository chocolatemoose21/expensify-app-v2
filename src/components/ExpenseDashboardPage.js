import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListItem from './ExpenseListItem';
import ExpenseListFilters from './ExpenseListFilters';



const ExpenseDashboardPage = () => (
  <div>
    <p> This is some text that works! </p>
    <ExpenseList/>
    <ExpenseListItem/>
    <ExpenseListFilters/>
  </div>
);

export default ExpenseDashboardPage;
