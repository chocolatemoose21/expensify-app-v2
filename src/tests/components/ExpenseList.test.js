import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render ExpenseList with expenses', () => {
  //Constants +
    //Call to render
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);

  //Assertion
  expect(wrapper).toMatchSnapshot();
});

test('should render expense list with empty message', () => {
  //Call to render
  const wrapper = shallow(<ExpenseList {...expenses[0]}/>);
  //Assertion
  expect(wrapper).toMatchSnapshot();
});
