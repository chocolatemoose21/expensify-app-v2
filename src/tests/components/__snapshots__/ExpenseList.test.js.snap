// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <h1>
     Expense List
  </h1>
  <ExpenseListItem
    amount={500}
    createdAt={0}
    description="Apple Sauce"
    id="1"
    key="1"
  />
  <ExpenseListItem
    amount={900}
    createdAt={-345600000}
    description="Gum"
    id="2"
    key="2"
  />
  <ExpenseListItem
    amount={400}
    createdAt={345600000}
    description="Soda"
    id="3"
    key="3"
  />
</div>
`;

exports[`should render expense list with empty message 1`] = `
<div>
  <h1>
     Expense List
  </h1>
  <ExpenseListItem
    amount={500}
    createdAt={0}
    description="Apple Sauce"
    id="1"
    key="1"
  />
  <ExpenseListItem
    amount={900}
    createdAt={-345600000}
    description="Gum"
    id="2"
    key="2"
  />
  <ExpenseListItem
    amount={400}
    createdAt={345600000}
    description="Soda"
    id="3"
    key="3"
  />
</div>
`;
